draft-ietf-quic-applicability.md

```diff
@@ -566,42 +590,49 @@ to the server instance. The server can provide an IPv4 and an IPv6 address in a
-during connection establishment and the effective value for this connection is
-the minimum of the two values advertised by client and server. An application
-therefore should be able to configure its own maximum value as well as have
-access to the computed minimum value for this connection. An application may
-adjust the maximum idle timeout for new connections based on the number of open
-or expected connections, since shorter timeout values may free-up memory more
-quickly.
-
-If an application desires to keep the connection open for longer than the
-announced timeout, it can send keep-alive messages; a QUIC implementation may
-provide an option to defer the time-out by sending keep-alive messages at the
-transport layer to avoid unnecessary load, as specified in {{Section 10.1.2 of
-QUIC}}. See {{resumption-v-keepalive}} for further guidance on keep-alives.
+# Connection Termination
+
+QUIC connections are terminated in one of three ways: implicit idle timeout,
+explicit immediate close, or explicit stateless reset.
+
+QUIC idle timeout is enabled via transport parameters. Client and server
+announce a timeout period and the effective value for the connection is the
+minimum of the two values. After the timeout period elapses, the connection is
+silently closed. An application therefore should be able to configure its own
+maximum value, as well as have access to the computed minimum value for this
+connection. An application may adjust the maximum idle timeout for new
+connections based on the number of open or expected connections, since shorter
+timeout values may free-up memory more quickly.
+
+As long as application data is exchanged, the idle timeout does not expire. If
+an application has no data to send but desires to keep the connection open for
+longer than the timeout period, it can send keep-alive messages. A QUIC
+implementation may provide an option to defer the time-out by sending keep-alive
+messages at the transport layer to avoid unnecessary load, as specified in
+{{Section 10.1.2 of QUIC}}. Alternatively, applications using QUIC could define
+their own mechanism, such as an application-layer ping, that achieves a similar
+result. See {{resumption-v-keepalive}} for further guidance on keep-alives.
+
+An immediate close is signalled by a CONNECTION_CLOSE frame (see
+{{error-handling}}). Immediate close causes all streams to become immediately
+closed. QUIC endpoints can manage the cumulative maximum number of streams they
+would allow to be opened using the MAX_STREAMS frames but there is no mechanism
+to reduce the value. An application that uses QUIC might commit to a number of
+openable streams but require the connection to be closed (for example, a
+scheduled maintenance period). Depending on how an application uses QUIC streams
+(see {{use-of-streams}}), abrupt closure of actively-used streams may be
+undesireable or detrimental. In contrast, waiting for an endpoint to exhaust the
```

Comment:
This bit about streams completely surprised me.  It's now into flow control and maximum stream limits, which really don't belong here.  If you like the text, I would suggest finding it another home.

Reply:
Yes, section 4 is on use of streams. Maybe we can just add another short subsection there to talk about the max stream limit.

Reply:
Will see what I can do. This text was really the main thing I was looking for in my issue #175 because shutdowns affect streams but QUIC doesn't talk about what termination of streams means for an app. I realise now that a subsection for stream limits and termination is probably useful, and then this part can cross-referencet to that as a "global reset of all streams" type thing